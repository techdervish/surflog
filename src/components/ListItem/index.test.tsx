import {
  expectWrappersTreeToMatchSnapshot,
  createShallowWrapper,
  mocks,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('ListItem component', () => {
  const mockOnClick = jest.fn();

  const wrapper = createWrapper({
    occurences: mocks.occurences,
    record: mocks.record,
    onClick: mockOnClick,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('should render necessary parts', () => {
    expect(wrapper.find('PrevNextControl')).toHaveLength(1);
    expect(wrapper.find('Time')).toHaveLength(1);
    expect(wrapper.find('PageIcon')).toHaveLength(1);
    expect(wrapper.find('Title')).toHaveLength(1);
    expect(wrapper.find('Content')).toHaveLength(1);
  });

  it('should call onClick prop on click', () => {
    wrapper
      .find('a')
      .first()
      .simulate('click');
    expect(mockOnClick).toBeCalled();
  });
});
