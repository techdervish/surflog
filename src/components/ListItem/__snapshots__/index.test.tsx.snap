// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders the expected snapshot 1`] = `
<div
  className="root pv2 relative z-0"
  style={
    Object {
      "overflowWrap": "break-word",
    }
  }
>
  <div
    className="actions absolute fw4 absolute--fill white"
  >
    <a
      className="h-100 db pointer"
      onClick={[MockFunction]}
    />
    <PrevNextControl
      current={0}
      onClickBack={[Function]}
      onClickNext={[Function]}
      total={2}
    />
    <Time
      date={1565498007441}
    />
  </div>
  <div
    className="fw4 flex"
  >
    <PageIcon
      record={
        Object {
          "content": "·59mI enjoy writing tests using @kentcdodds method ((link: https://kentcdodds.com/blog/testing-implementation-details) kentcdodds.com/blog/testing-i…)
I did this with React, and now applying it to backend with serverless.Testing Implementation DetailsTesting implementation details is a recipe for disaster. Why is that? And what does it even mean?kentcdodds.com127",
          "id": "/rohitfoss00/status/1160393803703832576",
          "lastVisitedTime": 1565498007441,
          "link": "https://twitter.com//rohitfoss00/status/1160393803703832576",
          "page": "twitter",
          "title": "Rohit@rohitfoss00",
        }
      }
    />
    <Title
      occurences={
        Array [
          Object {
            "key": "content",
            "position": Array [
              6,
              5,
            ],
          },
          Object {
            "key": "content",
            "position": Array [
              6,
              5,
            ],
          },
        ]
      }
      record={
        Object {
          "content": "·59mI enjoy writing tests using @kentcdodds method ((link: https://kentcdodds.com/blog/testing-implementation-details) kentcdodds.com/blog/testing-i…)
I did this with React, and now applying it to backend with serverless.Testing Implementation DetailsTesting implementation details is a recipe for disaster. Why is that? And what does it even mean?kentcdodds.com127",
          "id": "/rohitfoss00/status/1160393803703832576",
          "lastVisitedTime": 1565498007441,
          "link": "https://twitter.com//rohitfoss00/status/1160393803703832576",
          "page": "twitter",
          "title": "Rohit@rohitfoss00",
        }
      }
    />
  </div>
  <div
    className="fw3"
    style={
      Object {
        "paddingTop": 4,
      }
    }
  >
    <Content
      index={0}
      occurence={
        Object {
          "key": "content",
          "position": Array [
            6,
            5,
          ],
        }
      }
      record={
        Object {
          "content": "·59mI enjoy writing tests using @kentcdodds method ((link: https://kentcdodds.com/blog/testing-implementation-details) kentcdodds.com/blog/testing-i…)
I did this with React, and now applying it to backend with serverless.Testing Implementation DetailsTesting implementation details is a recipe for disaster. Why is that? And what does it even mean?kentcdodds.com127",
          "id": "/rohitfoss00/status/1160393803703832576",
          "lastVisitedTime": 1565498007441,
          "link": "https://twitter.com//rohitfoss00/status/1160393803703832576",
          "page": "twitter",
          "title": "Rohit@rohitfoss00",
        }
      }
    />
  </div>
  <style
    jsx={true}
  >
    
          .root {
            transition: background 0.3s ease;
            border-top: 1px solid #eceef1;
          }
          .root .actions {
            display: none;
          }
          .root:hover {
            z-index: 1;
            box-shadow: 0 0 1px #010324b3;
            background: rgba(1, 3, 36, 0.07);
            margin: 0 -16px;
            padding-left: 16px;
            padding-right: 16px;
          }
          .root:hover .actions {
            display: block;
          }
          .actions {
            top: -2px;
            user-select: none;
          }
        
  </style>
</div>
`;
